src/otx/algo/common/utils/bbox_overlaps.py

```diff
@@ -142,15 +144,24 @@ def bbox_overlaps(
         >>> assert tuple(bbox_overlaps(nonempty, empty).shape) == (1, 0)
         >>> assert tuple(bbox_overlaps(empty, empty).shape) == (0, 0)
     """
+    if not (bboxes1.size(-1) == 4 or bboxes1.size(0) == 0) or not (bboxes2.size(-1) == 4 or bboxes2.size(0) == 0):
```

Comment:
`bboxes1.size(0) == 0` is to check if there is at least one bounding box, right?
If so, I think a message that can describe this must be added as well.

Reply:
No. It allows empty bboxes. It's to check that bboxes either contains valid bounding boxes or is empty. If neither of these conditions is met, the it fails.

Reply:
I separate it to 2, to make it clear.
```python
if not (bboxes1.size(-1) == 4 or bboxes1.size(0) == 0):
        msg = "bboxes1 must have a last dimension of size 4 or be an empty tensor."
        raise ValueError(msg)
if not (bboxes2.size(-1) == 4 or bboxes2.size(0) == 0):
    msg = "bboxes2 must have a last dimension of size 4 or be an empty tensor."
    raise ValueError(msg)
```

Reply:
Ah, I was mistaken. Thanks for the update.